staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -219,7 +233,7 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 	}
 
 	if rules.WarnIfAllMissing && len(missingList) > 0 && len(kubeconfigs) == 0 {
-		klog.Warningf("Config not found: %s", strings.Join(missingList, ", "))
+		rules.Warner.Warn(fmt.Sprintf("Config not found: %s", strings.Join(missingList, ", ")))
```

Comment:
make this a typed error so that if we add more warnings in the future the handler can distinguish between them

for example:

```go
type MissingConfigError struct {
  Missing []string
}
func (c *ConfigNotFoundError) Error() string {
  return fmt.Sprintf("Config not found: %s", strings.Join(c.Missing, ", "))
}
```